MaraBot/Commands/LeaderboardCommandModule.cs

```diff
@@ -50,8 +60,33 @@ public async Task Execute(CommandContext ctx, int weekNumber = -1)
                 return;
             }
 
-            await Display.LeaderboardAsync(ctx, weekly, weekNumber == currentWeek);
-            await CommandUtils.SendSuccessReaction(ctx);
+            if (weekNumber == currentWeek)
+            {
+                var spoilerChannel = ctx.Guild.Channels
+                    .FirstOrDefault(channel => Config.WeeklySpoilerChannel.Equals(channel.Value.Name));
+
+                if (spoilerChannel.Value == null)
+                {
+                    await ctx.RespondAsync(
+                        "No spoiler channel set.\n" +
+                        "This shouldn't happen! Please contact your friendly neighbourhood developers!");
+                }
```

Comment:
Maybe make this a helper function too.
Let the function throw in the end if it can't find the role so this error also ends up in the logs.
We can also apply this to the role retrieval.

Reply:
Fixed in a55a610d2e1c35881730d1d53e07e950a410593e.